app/src/main/java/com/kylecorry/trail_sense/settings/SettingsFragment.kt

```diff
@@ -143,6 +136,18 @@ class SettingsFragment : PreferenceFragmentCompat() {
             R.id.action_action_settings_to_diagnosticFragment
         )
 
+        onClick(switch(R.string.pref_enable_experimental)) {
+            val pm: PackageManager? = context?.applicationContext?.packageManager
+            val compName = ComponentName(
+                getPackageName(requireContext()),
+                getPackageName(requireContext()) + ".AliasMainActivity"
+            )
+            pm?.setComponentEnabledSetting(
```

Comment:
I believe this logic should also be in the PackageReplaced receiver - to enable or disable it on install depending on the experimental setting

Reply:
I'm not sure how to do this, just put the code into this if block? 
https://github.com/kylecorry31/Trail-Sense/blob/0a594fd42a6df4dcc7d15348e6f9011065184a64/app/src/main/java/com/kylecorry/trail_sense/receivers/PackageReplacedReceiver.kt#L9-L11
In my tests, that's not working (nothing happens after installing the app). Another idea would be to run it if you start the app (on every start, in MainActivity.kt, or just if it's the first launch with a boolean value in the cache, this would be easier).

Reply:
Yes - that code doesn't always get run (only on package version upgrade)

Just in case it would be useful to also have it in MainActivity as you mentioned

If you want to just add it to MainActivity, I can handle the package replaced stuff

Reply:
Should be fixed in https://github.com/kylecorry31/Trail-Sense/pull/682/commits/ced19b3217911855100bd59f5e63e86a969b42a9